### ISSUE TITLE
Allow setting the OAuth flows per service

### ISSUE THREAD

**Issue by mayorova** [nmasse-itix/threescale-cicd#46]
3scale now allows setting allowed OAuth flows per service, that will be respected when creating clients in RH-SSO (done by Zync component).
JIRA: https://issues.jboss.org/browse/THREESCALE-774

It would be good to add this feature to the playbook and set the allowed flows based on the OpenAPI spec.
It seems that v2 of the spec only allows a single flow per security definition https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#security-scheme-object

The v3, however, accepts multiple flows https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#oauthFlowsObject

There is the following API endpoint to update this configuration in 3scale:
~~~
PATCH /admin/api/services/{service_id}/proxy/oidc_configuration.xml
~~~

<img width="770" alt="oidc" src="https://user-images.githubusercontent.com/1270328/54030619-75520280-41ac-11e9-8816-b0a037cd8744.png">

**Comment by nmasse-itix** [comment]
Thanks @mayorova ! 👍

Right now, the playbook only reads Swagger v2 but we could also implement the missing parts to read OAS v3 (and do the automatic conversion v3 -> v2 before uploading the Active Docs).

In both cases, we can take the proper value from the `securityScheme` object and let the user override this with an extra var that could be a string (single flow) or an array (multiple flows). This way we can support multiple flows with Swagger 2 and OAS 3.

**Comment by nmasse-itix** [comment]
This issue will become a top priority since the new behavior of the 3scale SaaS broke the smoketests with OIDC due to the new Zync mechanism.

**Comment by nmasse-itix** [comment]
Implemented in the latest release (1.2.0)

**Comment by mayorova** [comment]
Thanks a lot @nmasse-itix ! :)
